### ISSUE TITLE
Native lightGBM does not reject predictions for invalid features, can domain decorators solve this problem?

### ISSUE THREAD

**Issue by ToUranus** [jpmml/sklearn2pmml#338]
The LightGBM based on the Sklearn interface cannot receive string features, so a LableEncoder needs to be performed on discrete features. When there is an invalid feature model in the test set, the model will reject the prediction. Is there any way to solve this problem? I took a look at the handling of invalid features (such as treating invalid features as missing values) here in the domain decorator, but it doesn't seem to work. Can you help me out, thank you.

**Comment by vruusmann** [comment]
If you're embedding an LightGBM estimator into a Scikit-Learn pipeline, then the LightGBM estimator "inherits" feature information from Scikit-Learn (and loses its embedded feature information).

Therefore, any model schema customizations and relaxations made in Scikit-Learn pipeline should have full effect.

> I took a look at the handling of invalid features (such as treating invalid features as missing values) here in the domain decorator, but it doesn't seem to work

Perhaps you should show your code?

A brief introduction to feature specifications with the help of SkLearn2PMML decorator classes:
https://openscoring.io/blog/2020/02/23/sklearn_feature_specification_pmml/

The following should work - "accept invalid values as-is":

```python
from sklearn2pmml.preprocessing import CategoricalDomain

mapper = DataFrameMapper([
  (["cat_col"], [CategoricalDomain(invalid_value_treatment = "as_is"), LabelEncoder()])
])
```

Reference:
https://github.com/jpmml/sklearn2pmml/blob/master/sklearn2pmml/decoration/__init__.py#L52-L54

**Comment by vruusmann** [comment]
If the problem persists, please comment with a reproducible code example.

Everything has been working flawlessly on my computer for 5+ years.
